src/encoder.ml

```diff
@@ -98,33 +99,40 @@ let encode ({lookup_table; next_seq; _} as encoder) {name; value; never_index} =
     match token with
     | Some token -> Never_index (token + 1)
     | None ->
-      match LookupTable.find_opt lookup_table name with
-      | Some map ->
+      match LookupTable.find lookup_table name with
+      | map ->
         Never_index (seq_to_index next_seq (snd (ValueMap.choose map)))
-      | None -> Never_index 0
+      | exception Not_found -> Never_index 0
   else
-    match token, LookupTable.find_opt lookup_table name with
-    | Some token, Some map ->
-      begin match ValueMap.find_opt value map with
-      | Some seq -> Index (seq_to_index next_seq seq)
-      | None -> find_token encoder no_index token name value
+    begin match token with
+    | Some token ->
+      begin match LookupTable.find lookup_table name with
+      |  map ->
```

Comment:
There is an extra space before `map`.

Reply:
ah indeed. removed `Some` and kept the space.